src/Avalonia.Styling/Styling/StyleInstance.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive.Subjects;
+using Avalonia.Animation;
+using Avalonia.Styling.Activators;
+
+#nullable enable
+
+namespace Avalonia.Styling
+{
+    /// <summary>
+    /// A <see cref="Style"/> which has been instanced on a control.
+    /// </summary>
+    internal class StyleInstance : IStyleInstance, IStyleActivatorSink
+    {
+        private readonly List<ISetterInstance>? _setters;
+        private readonly List<IDisposable>? _animations;
+        private readonly IStyleActivator? _activator;
+        private readonly Subject<bool>? _animationTrigger;
+        private bool _active;
+
+        public StyleInstance(
+            IStyle source,
+            IStyleable target,
+            IReadOnlyList<ISetter>? setters,
+            IReadOnlyList<IAnimation>? animations,
+            IStyleActivator? activator = null)
+        {
+            Source = source ?? throw new ArgumentNullException(nameof(source));
+            Target = target ?? throw new ArgumentNullException(nameof(target));
+            _activator = activator;
+
+            if (setters is object)
+            {
+                var setterCount = setters.Count;
+
+                _setters = new List<ISetterInstance>(setterCount);
+
+                for (var i = 0; i < setterCount; ++i)
+                {
+                    _setters.Add(setters[i].Instance(Target));
+                }
+            }
+
+            if (animations is object && target is Animatable animatable)
+            {
+                var animationsCount = animations.Count;
+
+                _animations = new List<IDisposable>();
```

Comment:
You are not giving an initial size for this list, it should be `animationsCount`.